scheduler/scheduler.go

```diff
@@ -88,6 +100,7 @@ type DriverConfig struct {
 	BindingPort      uint16                                // optional
 	PublishedAddress net.IP                                // optional
 	NewMessenger     func() (messenger.Messenger, error)   // optional
+	NewDetector      func() (detector.Master, error)       // optional
```

Comment:
I thought I had a reason the first time around when I wrote NewMessenger. Followed the same pattern here without much forethought. I'm open to refactoring this pattern, but please in a separate PR.